modules/png-extraction/ImageExtractor.py

```diff
@@ -25,6 +25,9 @@
 import pathlib
 configs = {}
 
+with open('config.json', 'r') as f:
+    niffler = json.load(f)
+    print_only_public_headers = bool(niffler['PublicHeadersOnly'])
```

Comment:
You are handling your newly introduced config differently from the existing configurations.

Only line 491 was needed. This 28 - 30 was unnecessary. Or am I missing something here?

Reply:
I want use  print_only_public_headers as global variable and I am not able to use it. So I write this code.
Can you help me to make it as globle varible by other method ?

Reply:
You could write the code to support private tags the same way we are toggling to common headers or not.



It will be like,

    print_images = bool(configs['PrintImages'])
    print_only_common_headers = bool(configs['CommonHeadersOnly'])
    print_only_public_headers = bool(niffler['PublicHeadersOnly'])

Reply:
Also, of course, perform a first test before the commit to ensure it works with both true and false values for the newly introduced attribute.

Reply:
I do it sir. But it not work.

Reply:
with Pool(core_count) as p:
    res= p.imap_unordered(extract_headers, enumerate(chunk))
    for i,e in enumerate(res):
        headerlist.append(e)
data = pd.DataFrame(headerlist)
logging.info('Chunk ' + str(i) + ' Number of fields per file : ' + str(len(data.columns)))

 Can you help me how to pass the argument "PublicHeadersOnly" in extract_headers function ? I'm not able to pass argument from here.

Reply:
lines 93 and 314 should have print_only_public_headers included.

    final_res = execute(pickle_file, dicom_home, output_directory, print_images, print_only_common_headers, depth,
                        processes, flattened_to_level, email, send_email, no_splits, is16Bit, png_destination,
        failed, maps_directory, meta_directory, LOG_FILENAME, metadata_col_freq_threshold, t_start)

def execute(pickle_file, dicom_home, output_directory, print_images, print_only_common_headers, depth,
            processes, flattened_to_level, email, send_email, no_splits, is16Bit, png_destination,
    failed, maps_directory, meta_directory, LOG_FILENAME, metadata_col_freq_threshold, t_start):
    
    
    
If you follow how we handle "print_only_common_headers," you will see this flag is the same.